benchexec/tools/coveriteam-verifier-validator.py

```diff
@@ -50,14 +51,17 @@ def determine_result(self, run):
         will print out the produced aftifacts.
         If more than one dict is printed, the first matching one.
         """
-        for line in run.output:
+        verdict = None
+        for line in reversed(run.output):
             line = line.strip()
-            if "verdict" in line:
+            if "'verdict':" in line and verdict is None:
                 # CoVeriTeam outputs benchexec result categories as verdicts.
                 try:
-                    d = ast.literal_eval(line)
-                    if isinstance(d, dict):
-                        return d.get("verdict", result.RESULT_ERROR)
-                except SyntaxError:
+                    verdict = re.search(r"'verdict': '([a-zA-Z\(\)\ ]*)'", line).group(1).lower()
```

Comment:
I guess it would be slightly less error-prone to avoid the duplication of `'verdict'` and just search all lines with the regexp, taking the group content it there is a match?

Reply:
Good idea, thanks!
I had performance issues in mind, but the regex is probably not that much slower than the 'in' check.

Does the current code match your proposal?

Reply:
Yes, thanks. The performance of a regexp is certainly acceptable, and my understanding of a decent regexp engine is that cases like this where there is a fixed prefix (`'verdict': '`) of a match are just as fast as a simple search for that string (though even if it weren't, I would consider this unnecessary micro-optimization).

But if you wish: an obvious performance improvement would be to move the `verdict_regex.search(line)` call into an `if verdict is None:` condition. (In modern Python we could use the walrus operator, but here we need a nested if.)

Reply:
Yeah, I thought about it. But I would prefer to keep the nesting-level flat and don't worry about performance, if you agree.
I'm looking forward to being able to use the walrus operator, though.